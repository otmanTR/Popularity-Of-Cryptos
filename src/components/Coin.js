import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Coin = ({
  name, price, symbol, image, id,
}) => (
  <Link to={`/coin/${id}`} className="link">
    <li className="coin">
      <img className="coinImage" src={image} alt={image} />
      <h2 className="coinName">
        {name}
        <br />
        {symbol}
      </h2>
      <p>
        {price}
        $
      </p>
    </li>
  </Link>
);

Coin.propTypes = {
  name: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};

export default Coin;
