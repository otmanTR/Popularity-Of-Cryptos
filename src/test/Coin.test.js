import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Coin from '../components/Coin';

describe('Coin Card Component', () => {
  test('component match with the snapshot', () => {
    const tree = renderer.create(<BrowserRouter><Coin /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
