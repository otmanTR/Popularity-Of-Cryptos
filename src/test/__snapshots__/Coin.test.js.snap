// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin Card Component component match with the snapshot 1`] = `
<a
  className="link"
  href="/coin/undefined"
  onClick={[Function]}
>
  <li
    className="coin"
  >
    <img
      className="coinImage"
    />
    <h2
      className="coinName"
    >
      <br />
    </h2>
    <p
      className="coinPrice"
    >
      price:
      $
    </p>
  </li>
</a>
`;
